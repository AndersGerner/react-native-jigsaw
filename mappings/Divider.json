[{"name":"Divider","tag":"Divider","category":"layout","description":"A horizontal line used to divide content","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Divider.png","supports_list_render":false,"props":{"color":{"label":"Color","editable":true,"required":false,"value":"divider","type":"color"},"height":{"label":"Height","description":"The height (thickness) of the divider. If not provided, defaults to StyleSheet.hairlineWidth.","editable":true,"required":false,"type":"number","min":1,"max":50,"step":1,"precision":0}},"layout":{"height":1}}]