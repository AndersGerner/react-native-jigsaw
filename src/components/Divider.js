/* @flow */

import * as React from "react";
import { StyleSheet, View } from "react-native";
import { withTheme } from "../core/theming";
import { FORM_TYPES, COMPONENT_TYPES } from "../core/component-types";
import type { Theme } from "../types";

type Props = {
  style?: any,
  /**
   * @optional
   */
  theme: Theme
};

class Divider extends React.Component<Props> {
  render() {
    const {
      style,
      theme: { colors }
    } = this.props;

    return (
      <View
        style={[{ backgroundColor: colors.divider }, styles.divider, style]}
      />
    );
  }
}

const styles = StyleSheet.create({
  divider: {
    height: StyleSheet.hairlineWidth
  }
});

export default withTheme(Divider);

export const SEED_DATA = [
  {
    name: "Divider",
    tag: "Divider",
    category: COMPONENT_TYPES.primitive,
    description: "A horizontal line used to divide content",
    preview_image_url:
      "{CLOUDINARY_URL}/Divider.png",
    supports_list_render: false,
    props: {},
    layout: {
      width: "100%",
      height: 1
    }
  }
];
