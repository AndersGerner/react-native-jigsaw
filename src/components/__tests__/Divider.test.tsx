import * as React from "react";
import renderer from "react-test-renderer";
import Divider from "../Divider";

it("renders divider", () => {
  const tree = renderer.create(<Divider />).toJSON();

  expect(tree).toMatchSnapshot();
});
