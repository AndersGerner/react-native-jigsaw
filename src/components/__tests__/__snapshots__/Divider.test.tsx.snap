// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders divider 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(234, 237, 242, 1)",
      },
      undefined,
      Object {
        "height": 0.5,
      },
    ]
  }
/>
`;
